### ISSUE TITLE
Deploy new loss function python bindings

### ISSUE THREAD

**Issue by MatthewDaggitt** [vehicle-lang/vehicle#575]
The things that need doing are:
- [x] Add front-end bindings for generating a loss function similar to `generate_loss_function` in `py-vehicle-lang-tensorflow` https://github.com/vehicle-lang/vehicle/blob/1a0773fbb24c4c91b65713da0a15efbe3193bbba/py-vehicle-lang-tensorflow/src/vehicle_lang_tensorflow/__init__.py#LL10C1-L10C1
- [x] Remove `py-vehicle-lang-tensorflow`
- [x] Rename `py-vehicle-lang` to `vehicle-python`
- [ ] Add a CHANGELOG entry
- [x] Make a release for 0.4.0

**Comment by wenkokke** [comment]
There's a small issue, which is that the previous loss function translation, as well as the current one, doesn't abstract the properties over the network, so the translation has to happen on every invocation.

**Comment by wenkokke** [comment]
See [this test](https://github.com/vehicle-lang/vehicle/blob/1a0773fbb24c4c91b65713da0a15efbe3193bbba/py-vehicle-lang/tests/test_loss_function_tensorflow.py#L25-L40) for current usage example.

@ndslusarz: The current way to use `vehicle_lang` for loss functions is using these two separate functions:
- [`vehicle_lang.session.load`](https://github.com/vehicle-lang/vehicle/blob/1a0773fbb24c4c91b65713da0a15efbe3193bbba/py-vehicle-lang/src/vehicle_lang/session.py#L111)
- [`TensorflowTranslation().compile`](https://github.com/vehicle-lang/vehicle/blob/1a0773fbb24c4c91b65713da0a15efbe3193bbba/py-vehicle-lang/src/vehicle_lang/loss_function/translation/tensorflow.py#L12) (which is inherited from [`PythonTranslation().compile`](https://github.com/vehicle-lang/vehicle/blob/1a0773fbb24c4c91b65713da0a15efbe3193bbba/py-vehicle-lang/src/vehicle_lang/loss_function/translation/python.py#L43))

For a minor improvement in efficiency, we could probably decompose `compile` into `translate` and `exec`, so that the Vehicle AST to Python AST translation doesn't run on every iteration. However, the real solution is to translate the Vehicle property to a function which takes the network as an input.

**Comment by MatthewDaggitt** [comment]
> There's a small issue, which is that the previous loss function translation, as well as the current one, doesn't abstract the properties over the network, so the translation has to happen on every invocation.

That's true, but I don't think it's a problem. The translation has to happen on every invocation of `generate_loss_function`, but not every time the loss function is called? The latter would be problematic, but the former isn't?

**Comment by wenkokke** [comment]
> > There's a small issue, which is that the previous loss function translation, as well as the current one, doesn't abstract the properties over the network, so the translation has to happen on every invocation.
> 
> That's true, but I don't think it's a problem. The translation has to happen on every invocation of `generate_loss_function`, but not every time the loss function is called? The latter would be problematic, but the former isn't?

At the moment, the translation wraps a lambda around each declaration:
https://github.com/vehicle-lang/vehicle/blob/1a0773fbb24c4c91b65713da0a15efbe3193bbba/py-vehicle-lang-tensorflow/src/vehicle_lang_tensorflow/__init__.py#L66

My translation doesn't do that at the moment, since it's not generally sound if, e.g., the declarations refer to each other. However, some solution is needed.

I've included the FreeNetworks translation, which gets the free network names in each declaration, which could be used to sensibly abstract over them.

However, none of this may be necessary. I don't know what guarantees the loss function backend offers w.r.t. declarations being fully inlined? If properties never refer to one another, then just abstracting over networks or creating a thunk is safe.

If they don't reference one another, what does the FreeName constructor do?
